### ISSUE TITLE
ActiveStorage - Safari requests the same image multiple times

### ISSUE THREAD

**Issue by matthee** [rails/rails#37766]
### Steps to reproduce

1. Setup ActiveStorage using the default Disk service
2. View an attached image in Safari and observe requests for the image in the server log on every page-view.

### Expected behavior

Similar to Firefox and Chrome, there should only be only one request to the server per image.

### Actual behavior

Safari requests the same image over and over again during consecutive page-views.

### Possible Fix

Contrarily to Firefox and Chrome, Safari does not respect the cache headers if the server responds with a redirect. Instead, Safari uses the cache headers of the request actually returning the image (which is served by the DiskController).

It is possible to achieve the same behavior in Safari, by setting Cache-Headers during the `DiskController#show` action with a call to `expires_in`:

```ruby
  before_action :set_expiry, only: :show
  
  private
  def set_expiry
    expires_in ActiveStorage.service_urls_expire_in
  end
```

This issue is also related to ActionText. Trix internally uses `cloneNode` to reuse image elements during editing. The clone is considered to be a completely new element as far as the browser is concerned, and the necessary steps to request the images are started. Since Safari does not respect the caching headers of the BlobController, it requests the image once again from the server, instead of serving it from the cache. 

However, the DiskController represents a different level of abstraction and should IMHO behave like any other CDN. Therefore, I am not sure if the proposal is the proper way of fixing this behavior.

### System configuration
**Rails version**: 6.0.1

**Ruby version**: 2.6.5

**Comment by matthee** [comment]
@georgeclaghorn thoughts?

I am also willing to create a PR to address this issue.

**Comment by javinto** [comment]
I would like to reopen this issue. I have experienced this problem with several applications running Rails 7.2 and now even Rails 8. The only solution until now I could find is to configure 

`config.active_storage.resolve_model_to_route= :rails_storage_proxy`

but this is not ideal. I experienced problems with downloading larger images. 

I just solved it for an application by applying the proposed solution above setting the expiration. But to do that I had to monkeypatch the ActiveStorage::BaseController. 

I know of other people experiencing this problem as well. A pull request would be welcome!
